### PR TITLE
Aws ECR resource

### PR DESCRIPTION
**Pull request by dmurray-chef**
Signed-off-by: Darren Murray <dmurray@chef.io>

### Description

Add Aws ECR resource, unit tests, integration test, tf and docs

### Issues Resolved

https://github.com/inspec/inspec-aws/issues/46

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: test/integration/verify/controls/aws_ecr.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+title 'Test single AWS ECR'
+
+aws_ecr_name = "aws-ecr-name-fhxglcpleahkwukwlzdyywblu"
+#aws_ecr_name = attribute(:aws_ecr_name, default: '', description: 'The AWS ECR name.')
```
Comment (r-fennell) [inline test/integration/verify/controls/aws_ecr.rb]: 👀

File: test/integration/verify/controls/aws_ecr.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+control 'aws-ecr-1.0' do
+
+  impact 1.0
+  title 'Ensure AWS ECR has the correct properties.'
+
+  describe aws_ecr(repository_name: aws_ecr_name) do
+    it                       { should exist }
+    its ('repository_name')  { should eq aws_ecr_name }
```
Comment (r-fennell) [inline test/integration/verify/controls/aws_ecr.rb]: Some controls around the `images` would be nice here.

Comment (dmurray-chef) [reply test/integration/verify/controls/aws_ecr.rb]: We currently don't have a good mechanism for pushing container images to ECR that would fit well into CI. We can revisit this later

File: test/integration/verify/controls/aws_ecr.rb (hunk 1 above)
Comment (dmurray-chef) [reply test/integration/verify/controls/aws_ecr.rb]: fixed

**Pull request merged**